### DISCUSSION TITLE
Mozilla.Firefox-installed/-preuninstall

### DISCUSSION THREAD

**Discussion by KnifMelti** [Romanitho/Winget-AutoUpdate#255] [MODs]
After install/upgrade has been confirmed:
- Remove the desktop link `"Firefox"`

Winget uninstall is what it is (depending on the manifest), hence **-preuninstall**:
- Kill the process `"firefox"`
- Silently Uninstall the NSIS (Nullsoft) `"Mozilla Firefox*"`

If left to the manifest uninstall method, the user is presented with a **Uninstall Wizard** during the uninstall where there's a checkbox (**Tell Mozilla why you uninstalled Firefox**) that if checked **could** lead to a security breach in **SYSTEM** context...
[Mozilla.Firefox-installed (-preuninstall).zip](https://github.com/Romanitho/Winget-AutoUpdate/files/10328167/Mozilla.Firefox-installed.-preuninstall.zip)

**Comment by danielaltamirano1993** [comment]
Really thanks for your work. I have to admit that using Brave Software is very safe. It has beaten Mozilla. But in software development I still use Mozilla 100%

**Reply by KnifMelti** [reply]
There's no single front-/back-end to rule them all 😉
I use **everything** in my line of work/private life, depending of what I need to have/accomplish, and that includes the big corporations solutions...
...this project is a way for me to in my final years of work (60 years old now) learn how to code in PS, and it's a fun way of translating my ideas/logic to reality (that can be used by everyone)!

**Reply by KnifMelti** [reply]
...so, thank @Romanitho for creating this **Repo**!

**Reply by Romanitho** [reply]
thank you for your hard work !
